src/view/com/lightbox/ImageViewing/components/ImageItem/ImageItem.android.tsx

```diff
@@ -52,106 +158,238 @@ const ImageItem = ({
+      const candidateScale = Math.max(imageAspect / screenAspect, screenAspect / imageAspect, MIN_DOUBLE_TAP_SCALE)
+      const maxScale = (imageDimensions.width / SCREEN.width) * MAX_ORIGINAL_IMAGE_ZOOM
+      const scale = Math.min(candidateScale, maxScale)
+      const candidateTransform = createTransform();
+      prependPinch(candidateTransform, scale, origin, {x: 0, y: 0})
+      const [dx, dy] = getPanAdjustment(candidateTransform)
+      const finalTransform = createTransform();
+      prependPinch(finalTransform, scale, origin, {x: dx, y: dy})
+      committedTransform.value = withClampedSpring(finalTransform)
+    });
+
+  const dismissSwipePan = Gesture.Pan()
+    .enabled(!isScaled)
+    .activeOffsetY([-10, 10])
+    .failOffsetX([-10, 10])
+    .maxPointers(1)
+    .onUpdate(e => {
+      dismissSwipeTranslateY.value = e.translationY
+    })
+    .onEnd((e) => {
+      if (Math.abs(e.velocityY) > 1000) {
+        dismissSwipeTranslateY.value = withDecay({ velocity: e.velocityY })
+        runOnJS(onRequestClose)();
+      } else {
+        dismissSwipeTranslateY.value = withSpring(0, {
+          stiffness: 700,
+          damping: 50
+        })
+      }
+    });
+
+  const isLoading = !isLoaded || !imageDimensions;
   return (
-    <ScrollView
-      ref={imageContainer}
-      style={styles.listItem}
-      pagingEnabled
-      nestedScrollEnabled
-      showsHorizontalScrollIndicator={false}
-      showsVerticalScrollIndicator={false}
-      contentContainerStyle={styles.imageScrollContainer}
-      scrollEnabled={swipeToCloseEnabled}
-      {...(swipeToCloseEnabled && {
-        onScroll,
-        onScrollEndDrag,
-      })}>
-      <AnimatedImage
-        {...panHandlers}
-        source={imageSrc}
-        style={imageStylesWithOpacity}
-        onLoad={onLoaded}
-        accessibilityLabel={imageSrc.alt}
-        accessibilityHint=""
-      />
-      {(!isLoaded || !imageDimensions) && <ImageLoading />}
-    </ScrollView>
-  )
+    <Animated.View
+      ref={containerRef}
+      style={styles.container}>
+      {isLoading && <ActivityIndicator size="small" color="#FFF" style={styles.loading} />}
+      <GestureDetector gesture={Gesture.Exclusive(consumeHScroll, dismissSwipePan, Gesture.Simultaneous(pinch, pan), doubleTap)}>
+        <AnimatedImage
+          source={imageSrc}
+          contentFit="contain"
+          style={[styles.image, animatedStyle]}
+          accessibilityLabel={imageSrc.alt}
+          accessibilityHint=""
+          onLoad={() => setIsLoaded(true)}
+        >
+        </AnimatedImage>
+      </GestureDetector>
+    </Animated.View>
+  );
 }
 
 const styles = StyleSheet.create({
-  listItem: {
-    width: SCREEN_WIDTH,
-    height: SCREEN_HEIGHT,
+  container: {
+    width: SCREEN.width,
+    height: SCREEN.height,
+    overflow: 'hidden'
   },
-  imageScrollContainer: {
-    height: SCREEN_HEIGHT * 2,
+  image: {
+    flex: 1,
   },
+  loading: {
```

Comment:
This is provided by RN if that's useful https://reactnative.dev/docs/stylesheet#absolutefill